openmc/mgxs/mgxs.py

```diff
@@ -1922,9 +1939,30 @@ def rxn_rate_tally(self):
     @correction.setter
     def correction(self, correction):
         cv.check_value('correction', correction, ('P0', None))
+
+        if correction == 'P0' and self.legendre_order > 0:
```

Comment:
Yes, yes they do. Sync'ing up between `Library` and `ScatterMatrixXS` would be challenging but doable. I've simply presumed that anyone using `Library` should employ the appropriate "safety measures" and avoid directly manipulating the `MGXS` objects in the `Library`, which is the objective of the `Library` class to some extent.